### ISSUE TITLE
Get `pkg test symbolic` working

### ISSUE THREAD

**Issue by cbm755** [cbm755/octsympy#1142]
I think there is a new `pkg test ...` feature upstream.  We should use it, speed up the deprecation of `octsympy_tests`.

I'd like to see it used in our CI.  C.f., #1141 and #1060.

**Comment by cbm755** [comment]
And just checked and this now works, presumably b/c of the #1060 fix.  So I guess we can use it in CI for example

**Comment by cbm755** [comment]
Still seems to run tests for all packages not just symbolic :(  Doesn't happen locally

**Comment by alexvong1995** [comment]
> Still seems to run tests for all packages not just symbolic :( Doesn't happen locally

Running `docker exec oc octave-cli --eval "pkg test symbolic"` in CI actually works!

```
  PASS                             2400
  FAIL                                0
  XFAIL (expected failure)           29
```

Maybe there's something wrong with the Makefile...

**Comment by alexvong1995** [comment]
Looks like an upstream bug to me. In m-file https://hg.savannah.gnu.org/hgweb/octave/file/04120d65778a/scripts/pkg/pkg.m, if we remove those `;` in `case "test"`..., we can see the value of the variables:

```
installed_pkgs_lst =
{
  [1,1] =
    scalar structure containing the fields:
      name = symbolic
      version = 2.9.0+
      date = 2021-10-16
      author = Colin B. Macdonald <cbm@m.fsf.org>
      maintainer = Colin B. Macdonald <cbm@m.fsf.org>
      title = Octave Symbolic Package using SymPy
      description = Adds symbolic calculation features to GNU Octave.  These  include common Computer Algebra System tools such as algebraic operations,  calculus, equation solving, Fourier and Laplace transforms, variable  precision arithmetic and other features.  Compatibility with other symbolic  toolboxes is intended.
      categories = symbolic
      url = https://octave.sourceforge.io/symbolic
      depends =
      {
        [1,1] =
          scalar structure containing the fields:
            package = octave
            operator = >=
            version = 4.2
      }
      systemrequirements = python, sympy (>= 1.2), mpmath (>= 1.0)
      license = GPL-3.0-or-later
      dir = /usr/share/octave/packages/symbolic-2.9.0+
      archprefix = /usr
      loaded = 1
}
Testing functions in package 'symbolic':
installed_pkgs_dirs =
{
  [1,1] = /usr/share/octave/packages/symbolic-2.9.0+
  [1,2] = /usr
}
installed_pkgs_dirs =
{
  [1,1] = /usr/share/octave/packages/symbolic-2.9.0+
  [1,2] = /usr
}
installed_pkgs_dirs =
{
  [1,1] = /usr
  [1,2] = /usr/share/octave/packages/symbolic-2.9.0+
}
```

Here `/usr` is causing `pkg test` to look for m-file in `/usr`, which is why tests outside of symbolic run.

Looking at the code, it appears m-file https://hg.savannah.gnu.org/hgweb/octave/file/04120d65778a/scripts/pkg/private/expand_rel_paths.m is causing the problem.

This part of the code
```
    if (strncmpi (pkg_list{i}.dir, "__OH__", 6))
      pkg_list{i}.dir = [ loc pkg_list{i}.dir(7:end) ];
      pkg_list{i}.archprefix = [ loc pkg_list{i}.archprefix(7:end) ];
```
never checks if `"__OH__"` is a prefix of `pkg_list{i}.archprefix`. It simply assume so. In our case, `pkg_list{i}.archprefix` is empty, causing `  loc = OCTAVE_HOME ();` which is `/usr`, causing `/usr` to be prepended to the empty path.

**Comment by cbm755** [comment]
nice work!  Can you file that upstream and link the issue number here?  I guess we'll not be porting our makefile to `pkg test` quite yet.

**Comment by alexvong1995** [comment]
Sure, the upstream bug is https://savannah.gnu.org/bugs/index.php?62681

**Comment by alexvong1995** [comment]
Fixes https://hg.savannah.gnu.org/hgweb/octave/rev/65c4d98352d3 and https://hg.savannah.gnu.org/hgweb/octave/rev/cf5f46b2e052 have been pushed to stable.

The status of https://savannah.gnu.org/bugs/index.php?62681 is now "Ready For Test".

**Comment by alexvong1995** [comment]
------- Original Message -------

**Comment by alexvong1995** [comment]
Upstream bug https://savannah.gnu.org/bugs/index.php?62681 fixed:

------- Original Message -------

**Comment by alexvong1995** [comment]
Fixed in 3070c07f79ff749158a3f3e239c467ddffec7787
